src/hotspot/share/opto/graphKit.cpp

```diff
@@ -3522,7 +3522,7 @@ Node* GraphKit::get_layout_helper(Node* klass_node, jint& constant_value) {
   const TypeKlassPtr* inst_klass = _gvn.type(klass_node)->isa_klassptr();
   if (!StressReflectiveCode && inst_klass != nullptr) {
     bool    xklass = inst_klass->klass_is_exact();
-    if (xklass || inst_klass->isa_aryklassptr()) {
+    if (xklass || (inst_klass->isa_aryklassptr() && inst_klass->is_aryklassptr()->elem() != Type::BOTTOM)) {
```

Comment:
Can you rename `inst_klass` to `klass_ptr` because it is general klass. It is weird to check like `inst_klass->isa_aryklassptr()`.

Reply:
I pushed a new commit that includes the rename.